src/arrayMethodJoin.js

```diff
@@ -4,8 +4,29 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let resultStr = '';
+    let sep = separator;
+
+    if (sep === null) {
+      sep = String(sep);
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      let item = this[i];
+
+      if (item === undefined || item === null) {
+        item = '';
+      }
```

Comment:
You can just add separator to the main string and use continue key word for moving to the next iteration